### PR TITLE
[CodeGen][NewPM] Port EdgeBundles analysis to NPM

### PR DESCRIPTION
**Pull request by optimisan**

### CODE REVIEW

File: llvm/include/llvm/Passes/MachinePassRegistry.def [hunk 1]
```diff
@@ -97,6 +97,7 @@ LOOP_PASS("loop-term-fold", LoopTermFoldPass())
 // LiveVariables can be removed completely, and LiveIntervals can be directly
 // computed. (We still either need to regenerate kill flags after regalloc, or
 // preferably fix the scavenger to not depend on them).
+MACHINE_FUNCTION_ANALYSIS("edge-bundles", EdgeBundlesAnalysis())
```
Comment (paperchalice) [inline llvm/include/llvm/Passes/MachinePassRegistry.def:100]: May need a pinter pass call the `view` method to dump egds.

**Review by arsenm — APPROVED** [review]
lgtm with clang-format errors fixed

File: llvm/include/llvm/Passes/MachinePassRegistry.def (hunk 1 above)
Comment (optimisan) [reply llvm/include/llvm/Passes/MachinePassRegistry.def:100]: Can introduce in a new patch

**Comment by optimisan** [comment]
### Merge activity

* **Nov 22, 6:17 AM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/llvm/llvm-project/116616).

**Pull request merged**